test/unit/spec/util/jobLogViewerSpec.js

```diff
@@ -1,24 +1,178 @@
-define(['util/jobLogViewer', 'common/runtime'], (JobLogViewer, Runtime) => {
+define([
+    'bluebird',
+    'util/jobLogViewer',
+    'common/runtime',
+    'common/jobs',
+    '/test/data/jobsData',
+    'testUtil',
+], (Promise, JobLogViewer, Runtime, Jobs, JobsData, TestUtil) => {
     'use strict';
-    jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;
-    describe('Test the job log viewer module', () => {
-        const cssBaseClass = 'kb-log';
-        let hostNode = null,
-            runtimeBus = null;
 
-        beforeAll(() => {
-            window.kbaseRuntime = null;
+    const cssBaseClass = JobLogViewer.cssBaseClass;
+
+    const jobsByStatus = {};
+    JobsData.allJobs.forEach((job) => {
+        jobsByStatus[job.status] = job;
+    });
+
+    const endStates = ['completed', 'error', 'terminated'];
+    const queueStates = ['created', 'estimating', 'queued'];
+
+    function getWidgetState(node) {
+        const stateNode = node.querySelector('[data-element="widget-state"]');
+        return JSON.parse(stateNode.textContent);
+    }
```

Comment:
this info shows up in developer mode; the widget is initialised in dev mode for all these tests.